source/Missions/Services/Agents/Handlers/AgentDamageHandler.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Serialization;
+using LiteNetLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Agents.Packets;
+using Missions.Services.Network;
+using System;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    /// <summary>
+    /// Handler for agent damage in a battle
+    /// </summary>
+    public interface IAgentDamageHandler : IHandler
+    {
+
+    }
+    /// <inheritdoc/>
+    public class AgentDamageHandler : IAgentDamageHandler
+    {
+        readonly INetworkAgentRegistry networkAgentRegistry;
+        readonly INetworkMessageBroker networkMessageBroker;
+        public AgentDamageHandler(INetworkAgentRegistry networkAgentRegistry, INetworkMessageBroker networkMessageBroker) 
+        {
+            this.networkAgentRegistry = networkAgentRegistry;
+            this.networkMessageBroker = networkMessageBroker;
+
+            networkMessageBroker.Subscribe<AgentDamage>(AgentDamageSend);
+            networkMessageBroker.Subscribe<NetworkAgentDamage>(AgentDamageRecieve);
+        }
+        ~AgentDamageHandler()
+        {
+            networkMessageBroker.Unsubscribe<AgentDamage>(AgentDamageSend);
+            networkMessageBroker.Unsubscribe<NetworkAgentDamage>(AgentDamageRecieve);
+        }
```

Comment:
Do the same for the other handlers in the PR as well
```suggestion
        ~AgentDamageHandler()
        {
            Dispose();
        }
        
        public void Dispose()
        {
            networkMessageBroker.Unsubscribe<AgentDamage>(AgentDamageSend);
            networkMessageBroker.Unsubscribe<NetworkAgentDamage>(AgentDamageRecieve);
        }
```